pkg/volume/csi/csi_plugin.go

```diff
@@ -482,20 +482,16 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 	var spec *volume.Spec
 	inlineEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume)
 
-	if inlineEnabled {
-		mode := driverMode(volData[volDataKey.driverMode])
-		switch {
-		case mode == ephemeralDriverMode:
-			spec = p.constructVolSourceSpec(volData[volDataKey.specVolID], volData[volDataKey.driverName])
-
-		case mode == persistentDriverMode:
-			fallthrough
-		default:
-			spec = p.constructPVSourceSpec(volData[volDataKey.specVolID], volData[volDataKey.driverName], volData[volDataKey.volHandle])
-		}
-	} else {
-		spec = p.constructPVSourceSpec(volData[volDataKey.specVolID], volData[volDataKey.driverName], volData[volDataKey.volHandle])
+	// If inlineEnabled is true and mode is ephemeralDriverMode,
+	// use constructVolSourceSpec to construct volume source spec.
+	// If inlineEnabled is false or mode is persistentDriverMode,
+	// use constructPVSourceSpec to construct volume construct pv source spec.
+	if inlineEnabled && driverMode(volData[volDataKey.driverMode]) == ephemeralDriverMode {
```

Comment:
I support simplifying this code. However, I would recommend:
Break up the if as
```
if inlineEnabled {
    if driverMode(...) == ephemeralDriveMode{ ... }
    return spec, nil
}
spec = ...
```

Reply:
thanks, done!